rfc/0003-indexmanagement.md

```diff
@@ -174,105 +232,245 @@ For SDKs with configurable logging semantics, a mean of toggling a DEBUG log of
+to_build = mgr.n1ql_index_build_deferred()
 print(to_build)
-mgr.watch_indexes(to_build)
+mgr.n1ql_index_watch(to_build)
+mgr.n1ql_index_watch(['ix_fields'], watch_primary=True)
 ```
 
 In Python one can also pass `other_buckets=True` to `list_indexes()`
 which will cause it to enumerate other buckets as well, if it can access them.
 
+The method naming is by default `n1ql_index_XXX` because the other bucketmanager
+methods are in the form of `designXXX`. There are additional aliases so that
+`create_n1ql_index = n1ql_create_inxex`
+
+## Node.js
+
+```javascript
+var bucketManager = bucket.manager()
+
+bucketManager.createPrimaryIndex({
+  name: '',
+  ignoreIfExists: false,
+  deferred: false
+}, function(err){})
+
+bucketManager.createIndex(NAME, FIELDS, {
+  ignoreIfExists: false,
+  deferred: false
+}, function(err){})
+
+bucketManager.dropPrimaryIndex({
+  name: '',
+  ignoreIfNotExists: false
+}, function(err){})
+
+bucketManager.dropIndex(NAME, {
+  ignoreIfNotExists: false
+}, function(err){})
+
+bucketManager.getIndexes(function(err, indexes){})
+
+bucketManager.buildDeferredIndexes(function(err, indexes){})
+
+bucketManager.watchIndexes(WATCH_LIST, {
+  timeout: 0
+}, function(err){})
+```
+
+## Go
+
+```go
+func (bm *BucketManager) CreatePrimaryIndex(customName string, ignoreIfExists, deferred bool) error {}
+
+func (bm *BucketManager) CreateIndex(indexName string, fields []string, ignoreIfExists, deferred bool) error {}
+
+func (bm *BucketManager) DropPrimaryIndex(customName string, ignoreIfNotExists bool) error {}
+
+func (bm *BucketManager) DropIndex(indexName string, ignoreIfNotExists bool) error {}
+
+func (bm *BucketManager) GetIndexes() ([]IndexInfo, error) {}
+
+func (bm *BucketManager) BuildDeferredIndexes() ([]string, error) {}
+
+func (bm *BucketManager) WatchIndexes(watchList []string, watchPrimary bool, timeout time.Duration) error {}
+```
+
+## C
+
+```c
+typedef struct {
+    /** Raw index definition */
+    const char *rawjson;
+    size_t nrawjson;
+    const char *name;
+    size_t nname;
+    const char *keyspace;
+    size_t nkeyspace;
+    /** Output parameter only. State of index */
+    const char *state;
+    size_t nstate;
+
+    /** Actual index text. For raw JSON use the `index_key` property */
+    const char *fields;
+    size_t nfields;
+
+    /**
+     * Modifiers for the index itself. This might be
+     * LCB_IXSPEC_F_PRIMARY if the index is primary. For raw JSON,
+     * use `"is_primary":true`
+     *
+     * For creation the LCB_IXSPEC_F_DEFER option is also accepted to
+     * indicate that the building of this index should be deferred.
+     */
+    unsigned flags;
+
+    /**
+     * Type of this index, Can be T_DEFAULT for the default server type, or
+     * an explicit T_GSI or T_VIEW.
+     * When using JSON, specify `"using":"gsi"`
+     */
+    unsigned ixtype;
+} lcb_N1INDEXSPEC;
+
+typedef struct {
+    lcb_N1INDEXSPEC spec;
+    lcb_N1IXMGMTCALLBACK callback;
+} lcb_CMDN1IXMGMT;
```

Comment:
@mnunberg 

File https://github.com/couchbase/libcouchbase/blob/2.6.0/include/libcouchbase/ixmgmt.h uses **ix** for index in name, so the structure in this spec, but released headers use just X, like `lcb_CMDN1XMGMT`
https://github.com/couchbase/libcouchbase/blob/2.6.0/include/libcouchbase/ixmgmt.h#L138

I think that the spec have to be corrected at least